h2o-core/src/main/java/water/parser/ParseDataset.java

```diff
@@ -774,7 +774,16 @@ private FVecParseWriter makeDout(ParseSetup localSetup, int chunkOff, int nchunk
       byte[] zips = vec.getFirstBytes();
       ZipUtil.Compression cpr = ZipUtil.guessCompressionMethod(zips);
```

Comment:
Do we really need to fetch these first bytes just to determine compression ?

Reply:
I think so: magic number is common way to identify file type

Reply:
although usually only first FEW bytes are needed :)

Reply:
If you see the IO methods I used in my new pieces of code in this class, maybe then you'll ask the same question or see an obvious error in my thinking. :)

Reply:
if you mean that it could now be moved just before L784 instead, then apparently yes.
but it's a slightly different question from the one you asked, sorry for not reading your mind ;)

Reply:
The answer is no we don't need to fetch these first bytes to determine first compression, as those bytes are fetched anyway in the coder later. But the IO of the old part would need to work like my new part then. This is what I meant.